pkg/rselection/impls/pgx/leader_test.go

```diff
@@ -363,6 +364,84 @@ func (s *LeaderSuite) TestPingNodes(c *check.C) {
 	c.Check(leader.unsuccessfulPing(), check.Equals, true)
 }
 
+func (s *LeaderSuite) TestLeaderPingSelf(c *check.C) {
+	defer leaktest.Check(c)
+
+	channel := c.TestName()
+	// Use a real notifier to send the Ping Request notification.
+	realNotifier := &PgxPgNotifier{pool: s.pool}
+	// Use a fake notifier to record the Ping Response notification.
+	fakeNotifier := &dummyNotifier{}
```

Comment:
Writing a unit test for this was really tricky and I couldn't figure out how to use just the real notifier and call `leader.pingNodes()`. But all other tests use the fake notifier + real notifier as well, so I guess there's a reason why we have to use the fake notifier?